openstack_controller/datadog_checks/openstack_controller/components/compute.py

```diff
@@ -313,11 +318,13 @@ def _report_server_flavor(self, server, tags):
         )
         self.check.log.debug("flavor_metrics_and_tags: %s", flavor_metrics_and_tags)
         for metric, value in flavor_metrics_and_tags['metrics'].items():
-            self.check.gauge(metric, value, tags=tags + flavor_metrics_and_tags['tags'], hostname=server['id'])
+            self.check.gauge(metric, value, tags=tags + flavor_metrics_and_tags['tags'], hostname=server_id)
 
     @Component.http_error()
     def _report_server_diagnostics(self, server, tags):
-        server_id = server['id']
+        server_id = server.get('id')
+        if server_id is None:
+            return
```

Comment:
Can we debug log if the server id is not found?

Reply:
Sure! But it's more of a sanity check than anything else. That case should not occur in production.